src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -146,7 +152,8 @@ public void testFromXContent() throws IOException {
     @Test
     public void testToQuery() throws IOException {
         QueryParseContext context = new QueryParseContext(index, queryParserService);
-        assertLuceneQuery(this.testQuery, this.testQuery.toQuery(context));
+        context.setAllowUnmappedFields(true);
```

Comment:
Would `setMapUnmappedFieldAsString(true)` be okay here? not sure if this is considered "dynamic". Otherwise I get  "Strict field resolution and no field mapping can be found for the field with name [vdFWvqKy]" because I randomly generate fieldnames for this unit test and we don't want to have to create mappings because we dont index anything here.